### ISSUE TITLE
Adding spaces before/after string literals- everywhere

### ISSUE THREAD

**Issue by csopaki** [SAP/abap-cleaner#143]
Hello Colleagues,

I use the ABAP Cleaner daily, and set the automatic adding of spaces between character literals and parentheses.
There are though some other places, where it might be very handy to **add spaces around the character literals**:
- declarative statements (e.g. in class header - DATA, TYPES, ...)
- value assignments (standalone or in VALUE #(...) statement, if you have multiple fields within the parentheses.

### Code snippet:

```
CLASS myClass DEFINITION
  PUBLIC
  FINAL
  CREATE PUBLIC .

  PUBLIC SECTION.
  PROTECTED SECTION.
  PRIVATE SECTION.
    METHODS: myMethod.

    CONSTANTS: myconstant TYPE string VALUE'xyz'. " <<<=== declaration
ENDCLASS.


CLASS myClass IMPLEMENTATION.
  METHOD mymethod.    
    DATA mytext TYPE string VALUE'cde'. "<<<=== declaration

    DATA: somevariable TYPE string,
          BEGIN OF somedata,
            text   TYPE string,
            number TYPE i,
          END OF somedata,
          sometexts TYPE STANDARD TABLE OF string.

    somedata = VALUE #( text ='hsdfsd' number = 10 ). " <<<=== string literal in VALUE statement
    somevariable ='abc'. " <<<=== standalone string literal
  ENDMETHOD.

ENDCLASS.
```

Thanks a lot in advance, nice work you do with the tool, I like it very much!

**Comment by jmgrassau** [comment]
Hi Csopaki, 

thanks, great idea! Actually I think the cleanup rule "Standardize spaces next to parentheses" should be renamed to "Put spaces around string literals", because most of it is really about string literals, and the option "Remove multiple spaces from empty parentheses" could easily be handled by the rule "Remove needless spaces". 

Based on your examples, here are all cases that I found – they create warnings and even some red color in ADT, but are no syntax errors:

![image](https://github.com/SAP/abap-cleaner/assets/79639298/897b4d9f-b46b-4416-b301-935f4bd004d8)

So you could say, you always want spaces around string literals, except before the three operators : , . and except in cases where introducing a space would lead to a syntax error:

```ABAP
    " dynamic method calls 
    CALL METHOD lo_instance->('METHOD_NAME').
    CALL METHOD ('CLASS_NAME')=>('METHOD_NAME').

    " dynamic targets of component selectors
    ls_struc-('COMPONENT_NAME') = 1.
    lr_data_ref->('COMPONENT_NAME') = 1.

    " typed literals in ABAP SQL
    UPDATE demo_ddic_types
      SET int1 = int1`255`,
          int2 = int2`32767`,
          int4 = int4`2147483647`,
      WHERE id = char`Y`.
```

Hope I didn't miss anything? 

Kind regards,
Jörg-Michael

**Comment by jmgrassau** [comment]
Hi Csopaki, 

thanks a lot for opening this issue! I just released version 1.10.0 with the following changes:

* the option "Remove multiple spaces from empty parentheses" was moved to the cleanup rule "Remove needless spaces" (where it is activated if it was activated before)
* the cleanup rule "Standardize spaces next to parentheses" was renamed to "Put spaces around string literals", because that's what it now focuses on
* the following three options (highlighted in green) were added:
  ![image](https://github.com/SAP/abap-cleaner/assets/79639298/f2ba67a7-789a-4869-8164-bbc4435ef2fd)

Adding to the existing options, this now also cares for cases like these – some of which are marked as erroneous with red text color in ADT and create ATC findings, but could nevertheless be compiled:
![image](https://github.com/SAP/abap-cleaner/assets/79639298/0243e938-4bf6-4bd4-8013-a5ac5eff5e41)

Kind regards,
Jörg-Michael
